src/cost-explorer-mcp-server/awslabs/cost_explorer_mcp_server/server.py

```diff
@@ -0,0 +1,316 @@
+logger = logging.getLogger(__name__)
+
+# Initialize AWS Cost Explorer client
+ce = boto3.client('ce')
+
+
+class DateRange(BaseModel):
+    """Date range model for cost queries."""
+    start_date: str = Field(
+        ...,
+        description="The start date of the billing period in YYYY-MM-DD format. Defaults to last month, if not provided."
+    )
+    end_date: str = Field(
+        ...,
+        description="The end date of the billing period in YYYY-MM-DD format."
+    )
+
+
+class GroupBy(BaseModel):
+    """Group by model for cost queries."""
+    type: str = Field(
+        ...,
+        description="Type of grouping. Valid values are DIMENSION, TAG, and COST_CATEGORY."
+    )
+    key: str = Field(
+        ...,
+        description="Key to group by. For DIMENSION type, valid values include AZ, INSTANCE_TYPE, LEGAL_ENTITY_NAME, INVOICING_ENTITY, LINKED_ACCOUNT, OPERATION, PLATFORM, PURCHASE_TYPE, SERVICE, TENANCY, RECORD_TYPE, and USAGE_TYPE."
+    )
+
+
+class FilterExpression(BaseModel):
+    """Filter expression model for cost queries."""
+    filter_json: str = Field(
+        ...,
+        description="Filter criteria as a Python dictionary to narrow down AWS costs. Supports filtering by Dimensions (SERVICE, REGION, etc.), Tags, or CostCategories. You can use logical operators (And, Or, Not) for complex filters. Examples: 1) Simple service filter: {'Dimensions': {'Key': 'SERVICE', 'Values': ['Amazon Elastic Compute Cloud - Compute', 'Amazon Simple Storage Service'], 'MatchOptions': ['EQUALS']}}. 2) Region filter: {'Dimensions': {'Key': 'REGION', 'Values': ['us-east-1'], 'MatchOptions': ['EQUALS']}}. 3) Combined filter: {'And': [{'Dimensions': {'Key': 'SERVICE', 'Values': ['Amazon Elastic Compute Cloud - Compute'], 'MatchOptions': ['EQUALS']}}, {'Dimensions': {'Key': 'REGION', 'Values': ['us-east-1'], 'MatchOptions': ['EQUALS']}}]}."
+    )
+
+
+class CostMetric(BaseModel):
+    """Cost metric model."""
+    metric: str = Field(
+        "UnblendedCost",
+        description="The metric to return in the query. Valid values are AmortizedCost, BlendedCost, NetAmortizedCost, NetUnblendedCost, NormalizedUsageAmount, UnblendedCost, and UsageQuantity. Note: For UsageQuantity, the service aggregates usage numbers without considering units. To get meaningful UsageQuantity metrics, filter by UsageType or UsageTypeGroups."
+    )
+
+
+class DimensionKey(BaseModel):
+    """Dimension key model."""
+    dimension_key: str = Field(
+        ...,
+        description="The name of the dimension to retrieve values for. Valid values are AZ, INSTANCE_TYPE, LINKED_ACCOUNT, OPERATION, PURCHASE_TYPE, SERVICE, USAGE_TYPE, PLATFORM, TENANCY, RECORD_TYPE, LEGAL_ENTITY_NAME, INVOICING_ENTITY, DEPLOYMENT_OPTION, DATABASE_ENGINE, CACHE_ENGINE, INSTANCE_TYPE_FAMILY, REGION, BILLING_ENTITY, RESERVATION_ID, SAVINGS_PLANS_TYPE, SAVINGS_PLAN_ARN, OPERATING_SYSTEM."
+    )
+
+
+# Create FastMCP server
+app = FastMCP(title="Cost Explorer MCP Server")
```

Comment:
for consistency with the MCP servers in this repo, would change `app` to `mcp` (and update server.py accordingly) 
```suggestion
mcp = FastMCP(title="Cost Explorer MCP Server")
```